pkg/actions/toolkit/enable.go

```diff
@@ -0,0 +1,123 @@
+package toolkit
+
+import (
+	"context"
+
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/flux"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+)
+
+const allNamespaces = ""
+
+var defaultFluxComponents = []string{"helm-controller", "kustomize-controller", "notification-controller", "source-controller"}
+
+//go:generate counterfeiter -o fakes/fake_flux_client.go . FluxClient
+type FluxClient interface {
+	PreFlight() error
+	Bootstrap(opts *api.Toolkit) error
+}
+
+type Installer struct {
+	opts       *api.Toolkit
+	kubeClient kubeclient.Interface
+	fluxClient FluxClient
+}
+
+func New(k8sClientSet kubeclient.Interface, opts *api.Git) (*Installer, error) {
+	if opts.Toolkit == nil {
+		return nil, errors.New("expected git.toolkit in cluster configuration but found nil")
+	}
+
+	installer := &Installer{
+		opts:       opts.Toolkit,
+		kubeClient: k8sClientSet,
+		fluxClient: flux.NewClient(opts.Toolkit.AuthTokenPath, opts.Toolkit.GitProvider),
+	}
+
+	return installer, nil
+}
+
+func (ti *Installer) Run() error {
+	// TODO for now if we discover V1 components we just abort
+	// in future we may want to handle some migration magic
+	logger.Info("ensuring v1 repo components not installed")
+	v1Namespace, err := ti.checkV1()
+	if err != nil {
+		return errors.Wrap(err, "checking for flux v1 components")
+	}
+
+	if v1Namespace != "" {
+		logger.Warning("flux v1 components already installed in namespace %q. auto-migration not yet supported by eksctl. skipping installation", v1Namespace)
+		return nil
+	}
+
+	logger.Info("checking whether toolkit components already installed")
+	alreadyInstalled := ti.checkInstallation()
+	if alreadyInstalled {
+		logger.Warning("found existing Flux v2 components in namespace %q. skipping installation", ti.opts.Namespace)
+		return nil
+	}
+
+	logger.Info("running pre-flight checks")
+	if err := ti.fluxClient.PreFlight(); err != nil {
+		return errors.Wrap(err, "running Flux pre-flight checks")
+	}
+
+	logger.Info("bootstrapping gitops toolkit into cluster")
+	if err := ti.fluxClient.Bootstrap(ti.opts); err != nil {
+		return errors.Wrap(err, "running Flux Bootstrap")
+	}
+
+	logger.Success("GitOps Toolkit installed successfully")
+	logger.Success("see https://toolkit.fluxcd.io/ for usage instructions")
+
+	return nil
+}
+
+func (ti *Installer) checkV1() (string, error) {
+	deployments, err := ti.kubeClient.AppsV1().Deployments(allNamespaces).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, d := range deployments.Items {
+		if d.ObjectMeta.Name == "flux" {
+			return d.ObjectMeta.Namespace, nil
+		}
+	}
+
+	return "", nil
+}
+
+func (ti *Installer) checkInstallation() bool {
+	// TODO: this is the default set, we should maybe have an option to add whatever
```

Comment:
If we can punt on any the `TODO`s for this PR, should we leave them out of the code and make issues/track them in the main issue?

Reply:
👍  totally, is on my list to create issues for several "next-steps" and "missing pieces"